search/src/Components/upload/uploader_category_mapper.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import cloneDeep from 'lodash/cloneDeep';
+
+
+const useStyles = (theme) => ({
+    summary: {
+        marginLeft: '2.5em'
+    },
+    row: {
+        display: 'flex',
+        alignItems: 'baseline'
+    },
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+    },
+    span: {
+        marginRight: '1em'
+    },
+    applyButton: {
+        float: 'right'
+    }
+});
+
+class CategoryMapper extends React.Component {
+    
+    constructor(props) {
+        super(props);
+        this.state = {
+            categories: this.props.categories
+        }
+        this.modifyCategories = this.modifyCategories.bind(this);
+        this.changeCategory = this.changeCategory.bind(this);
+        this.changeSciName = this.changeSciName.bind(this);
+    }
+
+    modifyCategories(index, field, value) {
+        const newCategories = cloneDeep(this.state.categories);
+        newCategories[index][field] = value;
+        this.setState({categories: newCategories});
+    }
+
+    changeCategory(e, index) {
+        this.modifyCategories(index, "category", e.target.value);
+        this.props.handleCategoryStatus(false);
+        this.props.handleErrorMessage("init");
+    }
+
+    changeSciName(e, index) {
+        this.modifyCategories(index, "scitific_name", e.target.value);
+        this.props.handleCategoryStatus(false);
+        this.props.handleErrorMessage("init");
+    }
+
+    render() {
+        const { classes } = this.props;
+        return (
+            <React.Fragment>
+                <p className={classes.summary}>{this.state.categories.length} {this.state.categories.length > 1 ? 'categories' : 'category'} need mapping to weedcoco categories</p>
```

Comment:
This number should be updated as form values are filled in, such that if the category names are already valid weedcoco categories, this may start with "0 categories need mapping".

The 0 case should have plural "categories":
```suggestion
                <p className={classes.summary}>{this.state.categories.length} {this.state.categories.length != 1 ? 'categories' : 'category'} need mapping to weedcoco categories</p>
```